### PR TITLE
Add support for mocking procedures in mock connector

### PR DESCRIPTION
**Pull request by gaurav8297**

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/connector/MockConnector.java [hunk 1]
```diff
@@ -144,7 +146,8 @@
             BiFunction<ConnectorSession, ConnectorTableHandle, ConnectorTableProperties> getTableProperties,
             Supplier<Iterable<EventListener>> eventListeners,
             MockConnectorFactory.ListRoleGrants roleGrants,
-            MockConnectorAccessControl accessControl, Function<SchemaTableName, List<List<?>>> data)
+            MockConnectorAccessControl accessControl, Function<SchemaTableName, List<List<?>>> data,
```
Comment (kokosing) [inline core/trino-main/src/test/java/io/trino/connector/MockConnector.java:149]: please put each argument into separate line. (Actually it was my fault, I put `data` into the same line).

File: core/trino-main/src/test/java/io/trino/connector/MockConnectorFactory.java [hunk 2]
```diff
@@ -114,7 +116,8 @@ private MockConnectorFactory(
             BiFunction<ConnectorSession, ConnectorTableHandle, ConnectorTableProperties> getTableProperties,
             Supplier<Iterable<EventListener>> eventListeners,
             ListRoleGrants roleGrants,
-            MockConnectorAccessControl accessControl, Function<SchemaTableName, List<List<?>>> data)
+            MockConnectorAccessControl accessControl, Function<SchemaTableName, List<List<?>>> data,
```
Comment (kokosing) [inline core/trino-main/src/test/java/io/trino/connector/MockConnectorFactory.java:119]: same

File: core/trino-main/src/test/java/io/trino/connector/MockConnector.java [hunk 3]
```diff
@@ -64,6 +64,7 @@
 import io.trino.spi.connector.TopNApplicationResult;
```
Comment (kokosing) [inline core/trino-main/src/test/java/io/trino/connector/MockConnector.java:64]: Please add a test to `TestMockConnector` to make sure it works.

**Review by ksobolew — APPROVED** [review]
Looks good.

(I assume this will be used for something soon. Otherwise the YAGNI principle would apply :) )

**Comment by gaurav8297** [comment]
AC @kokosing

**Review by kokosing — APPROVED** [review]

**Pull request merged**